core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -137,6 +138,14 @@ class DAGScheduler(
   private[scheduler] val eventProcessLoop = new DAGSchedulerEventProcessLoop(this)
   taskScheduler.setDAGScheduler(this)
 
+  // Number of map, reduce tasks above which we do not assign preferred locations
+  // based on map output sizes.
+  private val SHUFFLE_PREF_MAP_THRESHOLD = 1000
```

Comment:
http://mail-archives.apache.org/mod_mbox/incubator-spark-dev/201402.mbox/%3CCAAsvFPk5rZfJv=Qr+Dufs=iKVvyrQmgPJY+pS_yTciuPeJvJLA@mail.gmail.com%3E explains it better than I can